torch/autograd/gradcheck.py

```diff
@@ -248,12 +248,12 @@ def fail_test(msg):
     for inp in tupled_inputs:
         if isinstance(inp, torch.Tensor):
             if inp.requires_grad:
-                if inp.dtype != torch.float64:
+                if inp.dtype != torch.float64 or inp.dtype != torch.complex128:
```

Comment:
This looks wrong no? It will always be True?
You want `not ( == or ==)` which translates to `(!= and !=)`.

Reply:
yeah you are right! thanks for the catch